test/integration/protocol_integration_test.cc

```diff
@@ -414,6 +414,23 @@ TEST_P(ProtocolIntegrationTest, MixedCaseScheme) {
   EXPECT_TRUE(scheme.empty() || scheme == "http");
 }
 
+TEST_P(ProtocolIntegrationTest, SchemeTransformation) {
+  config_helper_.addConfigModifier(
+      [&](envoy::extensions::filters::network::http_connection_manager::v3::HttpConnectionManager&
+              hcm) -> void {
+        hcm.mutable_scheme_header_transformation()->set_scheme_to_overwrite("https");
+      });
+
+  initialize();
+  codec_client_ = makeHttpConnection(lookupPort("http"));
+  default_request_headers_.setScheme("Http");
+  auto response =
+      sendRequestAndWaitForResponse(default_request_headers_, 0, default_response_headers_, 0);
+
+  auto scheme = upstream_request_->headers().getSchemeValue();
+  EXPECT_TRUE(scheme.empty() || scheme == "https");
```

Comment:
I would suggest
```
EXPECT_THAT(scheme, testing::AnyOf("", "https"));
```
because that way if the test fails you get useful information rather than "expected true but was false".

Reply:
(Also, a comment explaining why you can't just expect one value would be nice.)